atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/OracleTableUpdater.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.dbkvs.impl;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.nexus.db.sql.AgnosticResultSet;
+import com.palantir.nexus.db.sql.PalantirSqlConnection;
+
+public class OracleTableUpdater {
+    private final ConnectionSupplier conns;
+    private final TableReference tableRef;
+    private final TableNameGetter tableNameGetter;
+
+    public OracleTableUpdater(ConnectionSupplier conns, TableReference tableRef, TableNameGetter tableNameGetter) {
+        this.conns = conns;
+        this.tableRef = tableRef;
+        this.tableNameGetter = tableNameGetter;
+    }
+
+    public void update(Cell cell, long ts, byte[] oldValue, byte[] newValue) {
+        Object[] args = new Object[] {
+                cell.getRowName(),
+                cell.getColumnName(),
+                ts,
+                newValue,
+                cell.getRowName(),
+                cell.getColumnName(),
+                ts,
+                oldValue
+        };
+
+        String tableName = tableNameGetter.get(tableRef);
+        String sqlString = "/* UPDATE (" + tableName + ") */"
+                + " UPDATE " + tableName + ""
+                + " SET row_name = ?, col_name = ?, ts = ?, val = ?"
+                + " WHERE row_name = ?"
+                + " AND col_name = ?"
+                + " AND ts = ?"
+                + " AND val = ?";
+        int updated = ((PalantirSqlConnection) conns.get()).updateCountRowsUnregisteredQuery(sqlString,
+                args);
+        if (updated == 0) {
+            byte[] actualValue = getActualValue(cell, ts);
```

Comment:
In most cases, we do need this value - for example, if we're releasing and the lock is not as we expect, then that's a fairly severe issue.

If there's contention on acquire, then there might be less of a need to get the current value (because we expect sweep to have to wait for backup sometimes), but sometimes there will be, so we'd need an additional `get-current-lock` endpoint which the user would have to manually hit if they get tired of waiting.

Is that worth avoiding the perf hit of an extra few queries per minute?